lightning-invoice/src/payment.rs

```diff
@@ -264,13 +265,11 @@ pub trait Payer {
 	fn abandon_payment(&self, payment_id: PaymentId);
 }
 
-/// A trait defining behavior for routing an [`Invoice`] payment.
-pub trait Router {
-	/// Finds a [`Route`] between `payer` and `payee` for a payment with the given values.
-	fn find_route(
-		&self, payer: &PublicKey, route_params: &RouteParameters,
-		first_hops: Option<&[&ChannelDetails]>, inflight_htlcs: InFlightHtlcs
-	) -> Result<Route, LightningError>;
+/// A trait defining behavior for a [`Router`] implementation that also supports probing
+/// and scoring based on payment success data.
```

Comment:
Not really supporting probing but rather scoring channels based on probe success / failure.